om/include/OMR/Infra/Double.hpp

```diff
@@ -53,12 +53,12 @@ static constexpr bool isNaN(std::uint64_t value) {
 
 /// True for any quiet NaN.
 static constexpr bool isQNaN(std::uint64_t value) {
-  return isNaN(value) && areNoBitsSet(value, Double::NAN_SIGNAL_TAG);
+  return isNaN(value) && areAllBitsSet(value, Double::NAN_QUIET_TAG);
```

Comment:
Maybe we want to turn this into a single mask operation?
```C++
return areAllBitsSet(value, EXPONENT_MASK | NAN_QUIET_TAG);
```

same with the `isSNaN` below, but I guess it's a little more awkward. Plus, it looks like `isSNaN` will report true for infinity? That's probably bad... here is a fixed up version:

```C++
return
  isTagged(value, EXPONENT_MASK, EXPONENT_MASK | NAN_QUIET_TAG) &&
  areAnyBitsSet(value, NAN_EXTRA_BITS);
```

(as well, we might as well drop the `Double::*`, since we're in the `Double` namespace)

Reply:
The important part is the `isNaN(value)` check that runs first. It detects the difference between infinity and a NaN properly. I believe it makes it equivalent to what you have written.

I didn't want to use `return areAllBitsSet(value, EXPONENT_MASK | NAN_QUIET_TAG);` for `isQNaN()` because I liked the symmetry with the check below.  We still have to check the entire mantissa for anyBitSet in `isSNaN()`.  Maybe I can Godbolt this to see what this compiles to.

At the very least, the old code for `isSNaN()` was completely broken.

Reply:
Oh you're right, I thought `isNaN` would return `true` for infinity... for some reason :+1:
I'm okay with what you've got.